pymodbus/transport/serialtransport.py

```diff
@@ -8,25 +8,36 @@
 
 with contextlib.suppress(ImportError):
     import serial
+    import serial.rs485
 
 
 class SerialTransport(asyncio.Transport):
     """An asyncio serial transport."""
 
     force_poll: bool = False
 
-    def __init__(self, loop, protocol, *args, **kwargs) -> None:
+    def __init__(self, loop, protocol, rs485_settings, *args, **kwargs) -> None:
         """Initialize."""
         super().__init__()
         self.async_loop = loop
         self.intern_protocol: asyncio.BaseProtocol = protocol
-        self.sync_serial = serial.serial_for_url(*args, **kwargs)
+        self.sync_serial = self._serial_for_args(rs485_settings, *args, **kwargs)
         self.intern_write_buffer: list[bytes] = []
         self.poll_task: asyncio.Task | None = None
         self._poll_wait_time = 0.0005
         self.sync_serial.timeout = 0
         self.sync_serial.write_timeout = 0
 
+    def _serial_for_args(self, rs485_settings, *args, **kwargs) -> serial.Serial:
+        sync_serial: serial.Serial
+        if rs485_settings is not None:
```

Comment:
remove "is not None", that is the same but shorter.

Reply:
Can do - but while in this case it is the same - it's a pattern I like to avoid as sometimes things can evaluate to falsey when you don't expect them so.

Simple example:


```

maybe_bool: Optional[bool] = get_result()

if maybe_bool:
    print("no bool returned!")
else:
    ...
```